### ISSUE TITLE
Crash while calculate loss value with std::vector<label_t>

### ISSUE THREAD

**Issue by FohlenAFK** [tiny-dnn/tiny-dnn#992]
Hi guys,

`thiny-dnn/tiny_dnn/network.h` crashing while calculate the loss value:

```
  /**
   * calculate loss value (the smaller, the better)
   **/
  template <typename E>
  float_t get_loss(const std::vector<vec_t> &in,
                   const std::vector<label_t> &t) {
    float_t sum_loss = float_t(0);

    std::vector<tensor_t> label_tensor;
    normalize_tensor(t, label_tensor);

    for (size_t i = 0; i < in.size(); i++) {
      const vec_t predicted = predict(in[i]);
      for (size_t j = 0; j < predicted.size(); j++) {
        sum_loss += E::f(predicted, label_tensor[i][j]);
      }
    }
    return sum_loss;
  }
```

`const std::vector<label_t> &t` with values `0` and `1`
![image0](https://user-images.githubusercontent.com/28490587/45878358-02d99b80-bda1-11e8-980d-d9fe7d16b6a1.png)
![image1](https://user-images.githubusercontent.com/28490587/45878381-171d9880-bda1-11e8-9202-d5128799c4bd.png)

crashes on `sum_loss += E::f(predicted, label_tensor[i][j]);` because the `label_tensor` is not a 2D vector! So `j` should be always `0` in this case.

Anyone else with this problem?

Master: 1aec6a1ece0ba7a5e018a070bd52e045d49d1411
